syntax = "proto2";

package prop;

message msgLobby {
    required int32 uid = 1;
    required string name = 2;
	required int32 userNum = 3;
	required int32 useraMaxNum = 4;
	required bool closed = 5;
}

message msgRoom {
    required int32 uid = 1;
    required string name = 2;
	required int32 userNum = 3;
	required int32 useraMaxNum = 4;
}

message msgGame {
    required int32 uid = 1;
    required int32 number = 2;
	repeated msgUserInfo userList = 3;
}

message msgUserInfo {
    required int32 uid = 1;
    required string id = 2;
	required string Nickname = 3;
    optional int32 lobbyUID = 4;
    optional string lobbyName = 5;
    optional int32 roomUID = 6;
    optional string roomName = 7;
    optional int32 gameSessionUID = 8;
}

message msgUserInGameTotalInfo {
    required int32 userUid = 1;
	required float x = 2;
	required float y = 3;
	required float z = 4;
	required float qx = 5;
	required float qy = 6;
	required float qz = 7;
	required float qw = 8;
}

message msgUserInGamePosInfo {
    required int32 userUid = 1;
	required float x = 2;
	required float y = 3;
	required float z = 4;
	required float qx = 5;
	required float qy = 6;
	required float qz = 7;
	required float qw = 8;
}

message msgUserInGameStateInfo
{
    required int32 userUid = 1;
	required int32 stateBit32 = 2;
}

message accountCreateRequest
{
  required string id = 1;
  required string password = 2;
  required string nickname = 3;
}

message accountCreateResponse
{
	required bool success = 1;
}

message accountLoginRequest
{
  required string id = 1;
  required string password = 2;
}

message accountLoginResponse
{
	required bool success = 1;
}

message accountLogoutRequest
{
  required int32 empty = 1;
}

message accountLogoutResponse
{
	required bool success = 1;
}


message anyPlayerInfoSelfRequest
{
  required int32 empty = 1;
}

message anyPlayerInfoOtherRequestWithID
{
	required string otherID = 1;
}

message anyPlayerInfoOtherRequestWithNickName
{
	required string otherNickname = 1;
}

message anyPlayerInfoResponse
{
	required msgUserInfo userInfo = 1;
}


message anyWhisperRequestWithUID
{
	required int32 otherUID = 1;
	required string chat = 2;
}

message anyWhisperRequestWithID
{
	required string otherID = 1;
	required string chat = 2;
}

message anyWhisperRequestWithNickName
{
	required string otherNickname = 1;
	required string chat = 2;
}

message anyWhisperResponse
{
	required bool success = 1;
}


message outLobbyLobbylistRequest
{
	required int32 empty = 1;
}

message outLobbyLobbylistResponse
{
	required bool success = 1;
	repeated msgLobby lobbyList = 2;
}

message outLobbyEnterLobbyRequest
{
	required int32 lobbyUID = 1;
}

message outLobbyEnterLobbyResponse
{
	required bool success = 1;
	required int32 lobbyUID = 2;
}


message inLobbyPlayerlistRequest
{
	required int32 lobbyUID = 1;
}

message inLobbyPlayerlistResponse
{
	required bool success = 1;
	required int32 lobbyUID = 2;
	repeated msgUserInfo userList = 3;
}

message inLobbyRoomlistRequest
{
	required int32 lobbyUID = 1;
}

message inLobbyRoomlistResponse
{
	required bool success = 1;
	required int32 lobbyUID = 2;
	repeated msgRoom roomList = 3;
}


message inLobbyEnterRoomRequest
{
	required int32 roomUID = 2;
}

message inLobbyEnterRoomResponse
{
	required bool success = 1;
	required int32 roomUID = 2;
}

message inLobbyCreateRoomRequest
{
	required string roomName = 1;
	required int32 useraMaxNum = 2;
}

message inLobbyCreateRoomResponse
{
	required bool success = 1;
	required int32 roomUID = 2;
	required string roomName = 3;
}

message inLobbyLeaveLobbyRequest
{
	required int32 lobbyUID = 1;
}

message inLobbyLeaveLobbyResponse
{
	required bool success = 1;
	required int32 lobbyUID = 2;
}

message inLobbyChatRequest
{
	required int32 lobbyUID = 1;
	required string chat = 2;
}

message inLobbyChatResponse
{
	required bool success = 1;
}

message inLobbyChatBroadcast
{
	required int32 lobbyUID = 1;
	required string chat = 2;
}


message inRoomChatRequest
{
	required int32 roomUID = 1;
	required string chat = 2;
}

message inRoomChatResponse
{
	required bool success = 1;
}

message inRoomChatBroadcast
{
	required int32 roomUID = 1;
	required string chat = 2;
}

message inRoomLeaveRoomRequest
{
	required int32 roomUID = 1;
}

message inRoomLeaveRoomResponse
{
	required bool success = 1;
	required int32 roomUID = 2;
	required int32 lobbyUID = 3;
}

message inRoomForceOustRoomBroadcast
{
	required int32 roomUID = 1;
}

message inRoomReadyRequest
{
	required bool ready = 1;
}

message inRoomReadyResponse
{
	required bool success = 1;
}

message inRoomStartRequest
{
	required bool start = 1;
}

message inRoomStartResponse
{
	required bool success = 1;
}

message inRoomStartBroadcast
{
	required int32 inGameUID = 1;
}


message inGameSyncPlayerPosToServer
{
	required msgUserInGamePosInfo userInGameInfo = 1;
}

message inGameSyncPlayersPosFromServer
{
	repeated msgUserInGamePosInfo usersInGameInfo = 1;
}

message inGameSyncPlayerStateToServer
{
	required msgUserInGameStateInfo userStateInfo = 1;
}

message inGameSyncPlayersStateFromServer
{
	repeated msgUserInGameStateInfo userStateInfoList = 1;
}

message inGamePlayerTotalInfoRequest
{
	required int32 inGameUID = 1;
}

message inGamePlayerTotalInfoResponse
{
	required bool success = 1;
	optional msgUserInGamePosInfo usersInGameInfo = 2;
	optional msgUserInGameStateInfo userStateInfoList = 3;
}

message inGameAllPlayersTotalInfoRequest
{
  required int32 empty = 1;
}

message inGameAllPlayersTotalInfoResponse
{
	required bool success = 1;
	repeated msgUserInGamePosInfo usersInGameInfo = 2;
	repeated msgUserInGameStateInfo userStateInfoList = 3;
}


message inGameChatRequest
{
	required int32 inGameUID = 1;
	required string chat = 2;
}

message inGameChatResponse
{
	required bool success = 1;
}

message inGameChatBroadcast
{
	required string chat = 1;
}
